src/qt3utils/datagenerators/spectrometers/spectrometer.py

```diff
@@ -0,0 +1,230 @@
+
+        Parameters
+        ----------
+        acquisition_mode : Literal['single', 'step-and-glue', 'kinetic series', 'accumulation']
+            The acquisition mode.
+        **kwargs
+            Additional keyword arguments.
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The single or multiple spectra and the wavelength array.
+        """
+        if acquisition_mode not in self.ACQUISITION_MODES:
+            raise ValueError(f'Unsupported acquisition mode: {acquisition_mode}')
+        if acquisition_mode == 'single':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.single_acquisition, kwargs)
+            return self.single_acquisition(**valid_kwargs)
+        if acquisition_mode == 'step-and-glue':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.step_and_glue_acquisition, kwargs)
+            return self.step_and_glue_acquisition(**valid_kwargs)
+        if acquisition_mode == 'kinetic series':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.kinetic_series_acquisition, kwargs)
+            return self.kinetic_series_acquisition(**valid_kwargs)
+        if acquisition_mode == 'accumulation':
+            valid_kwargs = filter_only_valid_kwargs_for_method(self.accumulation_acquisition, kwargs)
+            return self.accumulation_acquisition(**valid_kwargs)
+
+    def single_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a single frame.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The single-frame spectrum and the wavelength array.
+        """
+        raise NotImplementedError()
+
+    def step_and_glue_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a single frame via the step-and-glue method.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The step-and-glued spectrum and the wavelength array.
+        """
+        raise NotImplementedError()
+
+    def kinetic_series_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a series of frames.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The series of spectra and the wavelength array.
+        """
+        raise NotImplementedError()
+
+    def accumulation_acquisition(self, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Method that acquires a single frame by accumulating multiple single frames and adds them together.
+
+        Returns
+        -------
+        Tuple[np.ndarray, np.ndarray]
+            The average spectrum and the wavelength array.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def stop_acquisition(self) -> None:
+        """
+        Stop/Pause the current acquisition.
+        """
+        raise NotImplementedError()
```

Comment:
@cmordi, if we make two separate abstract classes, these methods will be moved to the spectrometer DAQ class.

Reply:
Yes